### ISSUE TITLE
No resource for "enable secret" command and "iosxe_cli" not working

### ISSUE THREAD

**Issue by dga-nagra** [CiscoDevNet/terraform-provider-iosxe#120]
There is no resource for configuring `enable secret mysecret`. This should be a resource available.
I tried to find how to use `iosxe_restconf``resource but I was unable to find a decent documentation about how to use restconf...

### Workaround attempt (not working)
I finally decided to create a custom module using `iosxe_cli` this way:
```terraform
variable "secret" {
  type        = string
  description = "Secret to gain enable privileges"
  sensitive = true
}

resource "iosxe_cli" "example" {
  cli = <<-EOT
  enable secret ${var.secret}
  EOT
}
```

### Error output
Running it in debug mode produced the following error:
```C
Beginning to send CLI commands: tf_req_id=c469a2c6-3883-50c3-f99d-5c460e1687ba tf_resource_type=iosxe_cli @caller=github.com/CiscoDevNet/terraform-provider-iosxe/internal/provider/resource_iosxe_cli.go:95 @module=iosxe tf_provider_addr=registry.terraform.io/CiscoDevNet/iosxe tf_rpc=ApplyResourceChange timestamp="2024-02-27T11:01:38.168+0100"
2024/02/27 11:01:38 [DEBUG] HTTP Request: POST, https://10.11.12.13/restconf/operations/Cisco-IOS-XE-cli-rpc:config-ios-cli-rpc, {{"Cisco-IOS-XE-cli-rpc:input":{"config-clis":"enable secret mysecret\n"}}}
2024/02/27 11:01:40 [DEBUG] HTTP Response: {
  "errors": {
    "error": [
      {
        "error-message": "invalid path",
        "error-tag": "malformed-message",
        "error-type": "application"
      }
    ]
  }
}
2024/02/27 11:01:40 [ERROR] HTTP Request failed: StatusCode 400, RESTCONF errors {Error:[{ErrorType:application ErrorTag:malformed-message ErrorAppTag: ErrorPath: ErrorMessage:invalid path ErrorInfo:}]} {PatchId: GlobalStatus:{Ok:false Errors:{Error:[]}} EditStatus:{Edit:[]} Errors:{Error:[]}}
2024/02/27 11:01:40 [DEBUG] Exit from Do method
```

### Expected behaviour
1. A resource should exist for `enable secret` command
2. `iosxe_cli` should work

**Comment by dga-nagra** [comment]
Hi @danischm, I saw that you are the most active contributor and that you approved PR.
I am wondering if you could confirm this issue/have a workaround.
* How to run `enable secret mysecret` command
* How to make the resource `iosxe_cli` work

Nb: If needed, I may contribute.

**Comment by danischm** [comment]
We plan to add this to the `iosxe_system` resource. The `iosxe_cli` workaround should work with 17.13+, but apparently it is rejected with earlier versions of XE. What I would suggest to do instead is using the `iosxe_restconf` resource, which should allow you to configure *any* model using TF while a dedicated resource is not yet available.

**Comment by dga-nagra** [comment]
Hi, thank you a lot for the response !
> We plan to add this to the `iosxe_system` resource. The `iosxe_cli` workaround should work with 17.13+, but apparently it is rejected with earlier versions of XE. What I would suggest to do instead is using the `iosxe_restconf` resource, which should allow you to configure _any_ model using TF while a dedicated resource is not yet available.

I just took a look at [iosxe_system](https://registry.terraform.io/providers/CiscoDevNet/iosxe/latest/docs/resources/system). It already contains a lot of parameters and it won't allow to share part of the configuration easily.

In our case for example, we share most of the configuration, but we will have different enable secrets per device/region. 
Wouldn't it be better to simply have an "iosxe_enable" resource like

```terraform
resource "iosxe_enable" "enable" {
    secret = "mysecret"  # Required
    type    = "0"        # Optional
}
```
? It would also be easier to test and update. If you are okay with this, I could make you a PR.

for the `iosxe_cli` not working, I think it may indeed be related to the version.. I did the following command to get the list of available operations and did not see the one to rune CLI commands:
```bash
curl -k -u "${IOSXE_CREDS}" "https://${IOSXE_DEVICE_IP}/restconf/operations"
```
So I guess that we could put aside this `iosxe_cli` issue.

Let me know for the `enable secret` if `iosxe_enable` would be acceptable.

**Comment by danischm** [comment]
> I just took a look at [iosxe_system](https://registry.terraform.io/providers/CiscoDevNet/iosxe/latest/docs/resources/system). It already contains a lot of parameters and it won't allow to share part of the configuration easily.

Could you explain what you mean here?

**Comment by dga-nagra** [comment]
> Could you explain what you mean here?

I gave an example: We have a lot of shared configuration between our cisco devices, but some are made per-device, like the "enable secret". The reason for that is that we want to reduce the blast radius in case the password is leaked.

Another example is the "hostname" that is specific per host while we might want to have the same ip/log configuration across all devices.

If we want the same logging configuration on different devices you would have something like:
```terraform
resource "iosxe_system" "example" {
  hostname                    = "ROUTER-1"
  # ...
  login_delay                 = 10
  login_on_failure            = true
  login_on_failure_log        = true
  login_on_success            = true
  login_on_success_log        = true
  # ...
}

resource "iosxe_system" "example2" {
  hostname                    = "ROUTER-2"
  # ...
  login_delay                 = 10
  login_on_failure            = true
  login_on_failure_log        = true
  login_on_success            = true
  login_on_success_log        = true
  # ...
}
```

Or alternatively a bit better
```terraform
locals {
    devices = [
        {
          name = "ROUTER-1"
          # ...
        },
        {
          name = "ROUTER-2"
          # ...
        },
    ]
}
resource "iosxe_system" "example" {
  for_each = {for d in local.devices: d.name => d}
  hostname                    = each.key
  # ...
  login_delay                 = 10
  login_on_failure            = true
  login_on_failure_log        = true
  login_on_success            = true
  login_on_success_log        = true
  # ...
}
```

Now, let's image we want different "enable" secrets per devices based on various critierias (e.g. their regions), it will become even more complicated.

To sum this up, I just think that these definitions should not be coupled that much. It could be:
* iosxe_system_hostname
* iosxe_system_ip
* iosxe_system_login
* iosxe_system_enable (or just iosxe_enable)

This would be a lot more practical for complex configurations

**Comment by danischm** [comment]
You don't need to configure all attributes in a single resource, you can also have multiple instances of the same resource:
```hcl
resource "iosxe_system" "hostname" {
  hostname  = "Router1"
}

resource "iosxe_system" "enable_secret" {
  enable_secret = "MyPassword"
}
```

**Comment by dga-nagra** [comment]
@danischm Glad to know that.

I am not convinced this the way to go (SOLID pattern; resource overlap, need for prefixing, growing code for a single resource, ...). I would recommend having separate resources for it.

Would you appreciate a PR for an enable_secret resource ?
